splitter/splitter.go

```diff
@@ -229,14 +249,17 @@ func splitDefault(b LogEventBatch) []RSysLogMessage {
 // the subsequent decoding logic.
 func Split(b LogEventBatch) [][]byte {
 	var rsyslogMsgs []RSysLogMessage
-	var ok bool
-	if rsyslogMsgs, ok = splitAWSLambda(b); !ok {
-		if rsyslogMsgs, ok = splitAWSFargate(b); !ok {
-			rsyslogMsgs, ok = splitAWSBatch(b)
-			if !ok {
-				rsyslogMsgs = splitDefault(b)
-			}
-		}
```

Comment:
I kind of like the explicit boolean saying whether or not it matched... otherwise the matching logic bleeds into this split method. What do you think about something like

```golang
func stringify(rsyslogs []RSysLogMessage) [][]byte {
	out := make([][]byte, len(rsyslogs))
	for i := range rsyslogs {
		out[i] = []byte(rsyslogs[i].String())
	}
	return out
}

// Split takes a LogEventBatch and separates into a slice of enriched log lines
// Lines are enhanced by adding an Rsyslog prefix, which should be handled correctly by
// the subsequent decoding logic.
func Split(b LogEventBatch) [][]byte {
	if rsyslogMsgs, ok := splitAWSLambda(b); ok {
		return stringify(rsyslogMsgs)
	} else if rsyslogMsgs, ok := splitAWSFargate(b); ok {
		return stringify(rsyslogMsgs)
	} else if rsyslogMsgs, ok := splitAWSBatch(b); ok {
		return stringify(rsyslogMsgs)
	}
	return stringify(splitDefault(b))
}

```

Reply:
yeah I like that. Wasn't a fan of the increasingly indenting code and couldn't think of a great alternative.